packages/experience-builder-sdk/src/core/designTokenRegistry.ts

```diff
@@ -0,0 +1,14 @@
+import { OUTGOING_EVENTS } from '../constants';
+import { sendMessage } from '../communication/sendMessage';
+import { DesignTokensDefinition } from '../types';
+
+/**
+ * Register design tokens styling
+ * @param designTokenDefinition - {[key:string]: Record<string, string>}
+ * @returns void
+ */
+export const defineDesignTokens = (designTokenDefinition: DesignTokensDefinition) => {
+  sendMessage(OUTGOING_EVENTS.DesignTokens, {
+    designTokens: designTokenDefinition,
+  });
+};
```

Comment:
You will need to save the tokens into an object so that you can use them during rendering to resolve the actual values, similar to what we do for registered components: https://github.com/contentful/experience-builder/blob/f67f2f9b6ded47f860ee3404749fff88f96d8415/packages/experience-builder-sdk/src/core/componentRegistry.ts#L30.
I am not sure if this is in the scope of this PR though, feel free to disregard if it's gonna be done in a followup.

Reply:
Can you explain this part? So my understanding is that we need to save them on the user_interface side (which I did by using the immer building block store). This way, whenever we render the layout sidebar that uses the token key, we can pull the object from the store and then resolve the value at runtime.

Reply:
We'd need to resolve values both in the sidebar in user_interface and in the SDK side when rendering a component that uses a token. User interface could pass back to the SDK the actual value but that would work only for the editor mode. In the preview/delivery mode the SDK needs to also be able to resolve the values of the tokens directly since it cannot rely on user_interface.
For example in `preview`/`delivery` mode (and I think `editor` mode should work similarly) you would have an experience where somewhere in the `componentTree` would hold:
```
variableName: {
  type: "DesignToken",
  key: "padding.SM"
}
```
Since the actual value is not saved in the experience you'd need to look up that key into the DesignTokensRegistry.

That being said, this is more related to resolving the tokens than registering them so we could leave it out for now and add it when we implement that part.